backend/prompt_studio/prompt_studio_index_manager/prompt_studio_index_helper.py

```diff
@@ -18,46 +19,48 @@ def handle_index_manager(
         profile_manager: ProfileManager,
         doc_id: str,
     ) -> IndexManager:
-        document: DocumentManager = DocumentManager.objects.get(pk=document_id)
+        try:
 
-        index_id = "raw_index_id"
-        if is_summary:
-            index_id = "summarize_index_id"
+            with transaction.atomic():
```

Comment:
NIT: If this can be made as a decorator we could use it across functions easily rather than repeated nesting

Reply:
@johnyrahul also if my understanding of change done [here](https://github.com/Zipstack/unstract/pull/411/files#r1645895488) is correct, do we even need this atomic statement here?

Reply:
@chandrasekharan-zipstack  I believe decorator will work only on views I believe.

Reply:
@johnyrahul I didn't mean the decorator that django provides, I meant writing a decorator ourselves and using that for wrapping functions that need to be atomic

Reply:
@chandrasekharan-zipstack I don't think or own decorator will be much helpful. I think we can keep the current implementation as such.

Reply:
Discussed and closed as @ritwik-g  mentioned above.